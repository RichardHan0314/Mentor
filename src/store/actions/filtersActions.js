import { getAllFilters } from '../selectors';

export const GET_FILTERS = 'GET_FILTERS';
export const SET_ALL_FILTERS = 'SET_ALL_FILTERS';
export const SET_FILTERS = 'SET_FILTERS';

export const getFilters = (getState) => {
    const allFilters = getAllFilters(getState());
    return {
        type: GET_FILTERS,
        payload: allFilters,
    };
};

export const setAllFilters = () => (dispatch, getState) => {
    const allFilters = getAllFilters(getState());
    const newFilters = [...allFilters];
    newFilters.map(filter => ({ ...filter, status: false }));

    dispatch({
        type: SET_ALL_FILTERS,
        payload: { allFilters: newFilters, allFiltersChecked: true},
    });
};

export const setFilters = filterIndex => (dispatch, getState) => {
    const allFilters = getAllFilters(getState());
    allFilters[filterIndex].status = !allFilters[filterIndex].status;
    const newFilters = [...allFilters];

    dispatch({
        type: SET_FILTERS,
        payload: { allFilters: newFilters, allFiltersChecked: false},
    });
};
