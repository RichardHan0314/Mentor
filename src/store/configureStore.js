/* eslint-disable import/prefer-default-export */
import { createStore, applyMiddleware, compose } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { rootReducer} from '../reducers';
// import { ping } from './enhancers/ping'; // <-- подключаем наш enhancer

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(rootReducer, composeEnhancers(applyMiddleware(thunk, logger)));