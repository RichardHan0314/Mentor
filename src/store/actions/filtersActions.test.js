import configureStore from 'redux-mock-store';
import * as filtersActions from './filtersActions';

const mockStore = configureStore();
const store = mockStore();

describe('filters_actions', () => {
    beforeEach(() => {
        store.clearActions();
    });
  
    describe('resetFilters', () => {
        test('Dispatches the correct action and payload', () => {
            const expectedActions = [
                {'type': 'RESET_FILTERS'}
            ];
      
            store.dispatch(filtersActions.resetFilters());
            expect(store.getActions()).toEqual(expectedActions);
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('toggleFilter', () => {
        test('Dispatches the correct action and payload', () => {
            const filter = 'filterName';
            const expectedActions = [
                {
                    'payload': { filter },
                    'type': 'TOGGLE_FILTER',
                },
            ];
      
            store.dispatch(filtersActions.toggleFilter(filter));
            expect(store.getActions()).toEqual(expectedActions);
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});
