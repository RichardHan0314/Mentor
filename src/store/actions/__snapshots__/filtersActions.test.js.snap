// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters_actions resetFilters Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "RESET_FILTERS",
  },
]
`;

exports[`filters_actions toggleFilter Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "filter": "filterName",
    },
    "type": "TOGGLE_FILTER",
  },
]
`;
